src/data/moves/move-attrs/me-first-attr.ts

```diff
@@ -0,0 +1,62 @@
+import type { MoveConditionFunc } from "#app/@types/MoveConditionFunc";
+import type { Move } from "#app/data/moves/move";
+import { CallMoveAttr } from "#app/data/moves/move-attrs/call-move-attr";
+import type { Pokemon } from "#app/field/pokemon";
+import { globalScene } from "#app/global-scene";
+import type { BooleanHolder } from "#app/utils";
+import { BattlerTagType } from "#enums/battler-tag-type";
+import { MoveId } from "#enums/move-id";
+
+/**
+ * Attribute to copy the target's selected (and not yet used)
+ * move for the turn and use it against the target.
+ * Used for {@link https://bulbapedia.bulbagarden.net/wiki/Me_First_(move) | Me First}.
+ *
+ * NOTE: Unlike in mainline, Me First is currently redirected to the target's ally
+ * if the target faints or the target's index is otherwise vacant. This is intentional;
+ * we'd rather keep move behavior consistent than replicate what is likely a bug.
+ * @extends CallMoveAttr
+ */
+export class MeFirstAttr extends CallMoveAttr {
+  constructor() {
+    super();
+    this.hasTarget = true;
+  }
+
+  override apply(user: Pokemon, target: Pokemon, _move: Move, overridden: BooleanHolder): boolean {
+    const targetMove = this.getTargetMove(target);
+    if (targetMove) {
+      user.addTag(BattlerTagType.ME_FIRST_POWER_BOOST);
+      return super.apply(user, target, targetMove, overridden);
+    }
+    return false;
+  }
+
+  override getCondition(): MoveConditionFunc {
+    return (_user, target, _move) => {
+      const targetMove = this.getTargetMove(target);
+      return !!targetMove?.isAttackMove() && !invalidMeFirstMoves.has(targetMove.id);
+    };
+  }
+
+  protected getTargetMove(target: Pokemon): Move | undefined {
+    const { turnManager } = globalScene.currentBattle;
+    return turnManager.findCommandFromPokemon(target)?.turnMove?.move;
+  }
+}
+
+export const invalidMeFirstMoves: ReadonlySet<MoveId> = Object.freeze(
+  new Set([
+    MoveId.BEAK_BLAST,
+    MoveId.BELCH,
+    MoveId.CHATTER,
+    MoveId.COUNTER,
```

Comment:
```suggestion
    MoveId.COMEUPPANCE,
    MoveId.COUNTER,
```
Also, should all the G-Max moves be uncopiable by Me First? (Me First only existed up to gen 7, so Game Freak never dealt with this interaction.)

Reply:
Max moves should be handled the same as Gamefreak handled Z moves. If it can't copy Z moves, it shouldn't copy Max moves.

Reply:
but here G-Max moves are treated as normal moves, just that they get a bonus in damage when used by a max'd mon. No strong opinions one way or the other if they should be included

Reply:
They should not be included after https://github.com/Despair-Games/poketernity/pull/820/commits/37cda212ad76000677667088c942c603fd274263